lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -113,11 +113,19 @@ def serializable_hash_for_single_resource
         end
 
         def resource_identifier_type_for(serializer)
-          return serializer._type if serializer._type
-          if ActiveModelSerializers.config.jsonapi_resource_type == :singular
-            serializer.object.class.model_name.singular
+          value_or_block = serializer._type
```

Comment:
I think I'd rather see in implementation more like in https://github.com/rails-api/active_model_serializers/pull/1370/ and https://github.com/rails-api/active_model_serializers/pull/1383 and note https://github.com/rails-api/active_model_serializers/pull/1370/files#r48116422 hasn't yet be resolved.

cc @beauby 

Does that make sense? There's a lot going on in this method because there's uncertainty about how to handle the value being passed way down.